### DISCUSSION TITLE
Does nvim-tree have webicon for git sumodules?

### DISCUSSION THREAD

**Discussion by adimail** [nvim-tree/nvim-tree.lua#3034] [Q&A]
Hello,

I have been using nvim-tree for file management in my Neovim setup and I was wondering if there is any built-in support for displaying an icon for Git submodules in the tree view.

Currently, I see that nvim-tree supports various Git statuses like unstaged, staged, untracked, etc., but I couldn't find an option for a dedicated icon for Git submodules.

Is there any native support for this feature, or would it require custom configuration or plugin extensions to achieve this? If it's not currently supported, are there any plans to include this in the future?

Thank you!

**Comment by alex-courtis** [comment]
You are the first one with this idea.

Are you just after a different icon for `.gitmodules`? You can do that by submitting a PR to https://github.com/nvim-tree/nvim-web-devicons

If that's not it, please raise a [feature request](https://github.com/nvim-tree/nvim-tree.lua/issues/new?assignees=&labels=feature+request&projects=&template=feature_request.md&title=) and describe exactly the functionality you are after.

**Reply by adimail** [reply]
Yes, I would like to assign a specific icon for Git submodule folders in the directory structure. 

For example:

```lua
renderer = {
    -- other configurations...
    icons = {
        -- other icons...
        git = {
            -- existing git icons...
            submodule = '',  -- Git submodule icon
        },
    },
},
```

Could you guide me on how to implement this? If this is achievable, what steps would you recommend for setting it up or contributing to the project?
